hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1135,8 +1135,12 @@ int QuectelNcpClient::initReady(ModemState state) {
         skipAll(serial_.get(), 1000);
         CHECK(waitAtResponse(10000));
 
-        // Select either internal or external SIM card slot depending on the configuration
-        CHECK(selectSimCard());
+        // Cannot use ncpId() as this is not programmed on first boot during manufacturing
+        // Read NCP ID from OTP when initializing the modem after provisioning
```

Comment:
This is a unique case where a brand new device is powered on for the first time. There will be nothing in OTP so `platform_primary_ncp_identifier()` will return `PLATFORM_NCP_UNKNOWN`. 
Once OTP is programmed, we want to be able to power on the modem to collect EID. However, `conf.ncpIdentifier` will still be `PLATFORM_NCP_UNKNOWN`. We want to explicitly check OTP for BG95_S5 in order to avoid the CPIN errors and be able to turn the modem on

Reply:
I see.  So we will be updating the NCPID after `conf` is already created, so we will have to read the NCPID again directly.  I wonder if the MFG script should reset the device after programming the NCPID to make sure all other ncpId() code gets the most up to date value?

Reply:
It is not a script, it is a full running app. And we cannot reboot at this point because flash encryption has been enabled, but the encrypted MBR has not been programmed. The device will not boot if its reset at this point. The device must complete provisioning and report the details to the provisioning station so that the station can flash the encrypted MBR.